File: solutions/python/eliuds-eggs/7/eliuds_eggs.py
"""
Module for counting the number of 1s in the binary representation of an integer.
"""


def egg_count(display_value: int) -> int:
    """
    Counts the number of 1s in the binary representation of a given integer.

    Args:
        display_value (int): The integer to be evaluated.

    Returns:
        int: The count of 1s in the binary representation.
    """
    if display_value:
        return (display_value & 1) + egg_count(display_value // 2)
    return 0

File: solutions/javascript/eliuds-eggs/1/eliuds-eggs.js
export const eggCount = (displayValue) => {
  if (typeof displayValue !== 'number' || displayValue < 0 || !Number.isInteger(displayValue)) {
    throw new Error('Input must be a non-negative integer.');
  }

  let count = 0;
  while (displayValue > 0) {
    if (displayValue % 2 === 1) {
      count++;
    }
    displayValue = Math.floor(displayValue / 2);
  }
  return count;
};
